keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md

```diff
@@ -0,0 +1,999 @@
+
+[kubernetes#110782](https://github.com/kubernetes/kubernetes/issues/110782) tracks that lack of
+test coverage for OIDC, and [kubernetes#115122](https://github.com/kubernetes/kubernetes/pull/115122)
+attempts to rectify that gap.
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `k8s.io/kubernetes/pkg/kubeapiserver/options/authentication.go`: `2023-06-06` - `74.7`
+- `k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go`: `2023-06-06` - `0.0`
+- `k8s.io/apiserver/plugin/pkg/authenticator/token/oidc`: `2023-06-06` - `84`
+
+Note that as of 2023-06-06, the existing OIDC authenticator has no integration or e2e tests.
+
+Unit tests will be expanded to cover the new feature set of this KEP:
+
+- Structured config (including valdation)
+- CEL based expressions
+- Multiple client ID support
+- Discovery URL overrides
+- Automatic config reload
+- Multiple authenticators
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+Integration tests will cover parts of the new feature set as well:
+
+- CEL based expressions
+- Automatic config reload
+- Multiple authenticators
+
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+e2e tests will focus on testing a broad set of features together with "real" OIDC
+providers such as Okta, Azure AD, etc:
+
+- CEL based expressions
+- Multiple client ID support
+- Discovery URL overrides
+- Multiple authenticators
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Unit tests to validate CEL semantics
+- Unit tests for config validation
+- Initial e2e tests completed and enabled
```

Comment:
Any reason integration tests aren't included here? Is it because we need to test with a OIDC provider?

Reply:
This is a typo and should say integration tests, will fix.